src/ui/verificationSymbolStatusView.ts

```diff
@@ -0,0 +1,318 @@
+export default class VerificationSymbolStatusView {
+
+  public getFirstStatusForCurrentVersion(uriString: string): Promise<IVerificationSymbolStatusParams> {
+    return this.getFirstStatusForFilePromise(uriString).promise;
+  }
+
+  public static createAndRegister(
+    context: ExtensionContext,
+    languageClient: DafnyLanguageClient,
+    compilationStatusView: CompilationStatusView): VerificationSymbolStatusView {
+    const instance = new VerificationSymbolStatusView(context, languageClient, compilationStatusView);
+    window.onDidChangeActiveTextEditor(e => {
+      if(e !== undefined) {
+        const lastUpdate = instance.updatesPerFile.get(e.document.uri.toString());
+        if(lastUpdate !== undefined) {
+          instance.update(lastUpdate);
+        }
+      }
+    });
+    workspace.onDidChangeTextDocument(e => {
+      const uriString = e.document.uri.toString();
+      instance.updatesPerFile.delete(uriString);
+      instance.updateListenersPerFile.delete(uriString);
+    });
+    context.subscriptions.push(
+      languageClient.onVerificationSymbolStatus(params => instance.update(params))
+    );
+    return instance;
+  }
+
+  private getFirstStatusForFilePromise(uriString: string): ResolveablePromise<IVerificationSymbolStatusParams> {
+    let listener = this.updateListenersPerFile.get(uriString);
+    if(listener === undefined) {
+      let storedResolve: (value: IVerificationSymbolStatusParams) => void;
+      const promise = new Promise<IVerificationSymbolStatusParams>((resolve) => {
+        storedResolve = resolve;
+      });
+      listener = { resolve: storedResolve!, promise };
+      this.updateListenersPerFile.set(uriString, listener);
+    }
+    return listener;
+  }
+
+  private constructor(
+    private readonly context: ExtensionContext,
+    private readonly languageClient: DafnyLanguageClient,
+    private readonly compilationStatusView: CompilationStatusView) {
+    this.controller = this.createController();
+  }
+
+  private itemStates: Map<string, PublishedVerificationStatus> = new Map();
+  private itemRuns: Map<string, ItemRunState> = new Map();
+  private readonly runItemsLeft: Map<TestRun, number> = new Map();
+  private readonly updateListenersPerFile: Map<string, ResolveablePromise<IVerificationSymbolStatusParams>> = new Map();
+  private readonly updatesPerFile: Map<string, IVerificationSymbolStatusParams> = new Map();
+  private readonly controller: TestController;
+  private automaticRunEnd: boolean = false;
+  private processUpdateLock: Promise<void> = Promise.resolve();
+
+  private createController(): TestController {
+    const controller = tests.createTestController('verificationStatus', 'Verification Status');
+    controller.createRunProfile('Verify', TestRunProfileKind.Run, async (request) => {
+      const items: TestItem[] = this.getItemsInRun(request, controller);
+      const runningItems: TestItem[] = [];
+      let outerResolve: () => void;
+      await this.processUpdateLock;
```

Comment:
I think this will have the effect of letting the scheduler pick the next task before continuing. If this is intended, then you should name this `yieldAndRecoverLock()` as JavaScript is a single-threaded language.
Otherwise, can you please elaborate on why do you need this?